### ISSUE TITLE
installing on MacOS 11.3 fails as a result of out-of-date Test2

### ISSUE THREAD

**Issue by dbreiser** [finance-quote/finance-quote#184]
Not sure if this is really a Finance-Quote issue, but MacOS updates appear to wipe out anything user-installed in Perl.
After the 11.3 Big Sur update, I tried 'install Finance::Quote' in CPAN. It thrashed for a while with several 'make test NOT OK' messages and eventually failed.
One of the messages also said: Test2 version 1.302167 required--this is only version 1.302162
It was probably up the dependency tree a ways, but doing 'upgrade Test2' in CPAN followed by 'install Finance::Quote'  worked.

**Comment by mekanics** [comment]
Thx @dbreiser 🍺 
upgrading Test2 helped!

**Comment by bpschuck** [comment]
This is not an issue with Finance::Quote, but more of an issue with modules and module dependencies related to OSX 11.3 "Big Sur". Unless there are other objections I will be closing this PR in the next few days.

**Comment by bpschuck** [comment]
As indicated a few days ago, this issue is being closed.

**Comment by fasmat** [comment]
@dbreiser and @mekanics 

I'm struggling with installing Finance::Quote on Big Sur. I upgraded Test2 and ran gnc-fq-update but installation completes with the following statement:

```
All tests successful.
Files=56, Tests=82,  9 wallclock secs ( 0.12 usr  0.09 sys +  7.50 cusr  1.67 csys =  9.38 CPU)
Result: PASS
  BPSCHUCK/Finance-Quote-1.51.tar.gz
Tests succeeded but one dependency not OK (Test::Perl::Critic)
  BPSCHUCK/Finance-Quote-1.51.tar.gz
  [dependencies] -- NA
```

gnu-fq-check gives the following error:

```
You need to install the following Perl modules:
  Finance::Quote

Use your system's package manager to install them,
or run 'gnc-fq-update' as root.
missing-lib
```

And installing Perl::Critic gives me the following error:

```
Can't locate B/Keywords.pm in @INC (you may need to install the B::Keywords module) (@INC contains: inc /Library/Perl/5.30/darwin-thread-multi-2level /Library/Perl/5.30 /Network/Library/Perl/5.30/darwin-thread-multi-2level /Network/Library/Perl/5.30 /Library/Perl/Updates/5.30.2/darwin-thread-multi-2level /Library/Perl/Updates/5.30.2 /System/Library/Perl/5.30/darwin-thread-multi-2level /System/Library/Perl/5.30 /System/Library/Perl/Extras/5.30/darwin-thread-multi-2level /System/Library/Perl/Extras/5.30 .) at t/Variables/RequireLocalizedPunctuationVars.run.PL line 10.
```

B::Keywords is installed (at least `install B::Keywords` passes without any errors or warnings. I'm stuck here. Can one of you help? :-)

**Comment by bpschuck** [comment]
This is more an issue with your Perl configuration than it is a Finance::Quote problem.
You mentioned installing `Perl::Critic`, the module dependency is `Test::Perl::Critic`. Typo?
To test if `B::Keywords` and `Test::Perl::Critic` are properly installed you can try a couple different commands. Replace `<module>` with `B::Keywords` or `Test::Perl::Critic` in any of the following:
```
cpan -l | grep <module>
perldoc <module>
perl -M<module> -e1
```
Another possible cause on OSX could be that you have two different Perl installations. One is the distribution bundled with OSX and another installed by `Brew` or some other alternative. Typically CPAN modules installed in one of the distributions are not available in the other(s).

**Comment by bpschuck** [comment]
Another way to check for the modules is to execute a find for each path in Perl's `@INC`.
`for dir in $(perl -e "print qq(@INC)"); do find ${dir} -path '*/B/*' -type f -name 'Keywords.pm' 2>/dev/null; done`
and
`for dir in $(perl -e "print qq(@INC)"); do find ${dir} -path '*/Test/Perl/*' -type f -name 'Critic.pm' 2>/dev/null; done`

**Comment by fasmat** [comment]
Thanks for your input @bpschuck. I finally fixed the issue by doing the following:

On macOS Big Sur a lot of modules seem to be missing from the perl installation, and although ``B::Keywords`` installs successfully some tests for it are failing. So I installed it without executing tests:

```
sudo CPAN -T B::Keywords
```

after also upgrading Test2 with ``sudo perl -MCPAN -e 'upgrade Test2'`` i was finally able to install ``Finance::Quote`` by calling:

```
sudo perl -MCPAN -e 'install Finance::Quote'
```

Unfortunately that is not enough to actually use it. I was always receiving "Internal Server Errors" when trying to fetch quotes with GnuCash. The problem turned out to be missing certificates that I installed with

```
sudo cpan Mozilla::CA
```

Now finally I can fetch quotes again!

**Comment by vincentl** [comment]
@bpschuck - I'm wondering if we are missing some dependencies.  I'll try to do some testing with a fresh perl installation.

**Comment by bpschuck** [comment]
A fresh Perl installation on OSX Big Sur (aka 11.x instead of 10.16.x)? 
While we may be missing some dependencies (I added some for the v1.50 release), the last 3 releases of OSX have each introduced challenges with respect to compile tools and Perl. From Mojave to Catalina to Big Sur, I believe each of them moved around where additional header include files have been stored. Even with xCode installed, one usually had to also install an additional package that was included but not installed (on Mojave/10.14.x it was /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg ).
And like I explained to Matthias, some who also install Perl using HomeBrew have two different versions which can muddy the waters. I have run into issues where modules installed into the PERL5LIB tree will work with one Perl but not the other, as their dependencies may have been installed using CPAN but not in the PERL5LIB directory tree.
But, if you have a fresh Big Sur system, please test and add the dependencies to dist.ini.
Thanks.
